### PR TITLE
Fixed some issues with User-Role relationship & Adapted tests

### PR DESCRIPTION
**Pull request by Pasichniuk**

### CODE REVIEW

File: src/main/java/com/example/app/entity/Role.java [hunk 1]
```diff
@@ -25,6 +25,6 @@
     @Column(name = "role_name", unique = true, nullable = false)
     private String roleName;
 
-    @ManyToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL, mappedBy = "roles")
-    private Collection<User> users;
+    @OneToMany(fetch = FetchType.EAGER, mappedBy = "role")
+    private Set<User> users = new HashSet<>(0);
```
Comment (apokhyla-inc) [inline src/main/java/com/example/app/entity/Role.java:31]: Interesting...do we really need this inline initialization here?

Comment (Pasichniuk) [reply src/main/java/com/example/app/entity/Role.java:31]: Some cool guys from StackOverFlow said it's amazing solution!

Comment (apokhyla-inc) [reply src/main/java/com/example/app/entity/Role.java:31]: That's kinda strange...I'll double-check this now.

Comment (apokhyla-inc) [reply src/main/java/com/example/app/entity/Role.java:31]: Let's add TODO here and back to it later.

**Pull request merged**